cleverhans/max_confidence.py

```diff
@@ -0,0 +1,106 @@
+"""
+The MaxConfidence attack.
+
+An attack designed for use against models that use confidence thresholding
+as a defense.
+If the underlying optimizer is optimal, this attack procedure gives the
+optimal failure rate for every confidence threshold t > 0.5.
+
+Publication: https://openreview.net/forum?id=H1g0piA9tQ
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from six.moves import range
+import tensorflow as tf
+
+from cleverhans.attacks import Attack
+from cleverhans.model import Model
+from cleverhans.attacks import ProjectedGradientDescent
+
+
+class MaxConfidence(Attack):
```

Comment:
Is there a specific reason for including this attack outside of the file `attacks.py`?

Reply:
No, I originally had it in a separate file because I was developing it in my own repo, but it could be moved into the main attacks.py easily if that is where it goes. I hadn't move it anywhere specific yet because I'm not sure what's going on with attacks.py vs attacks_tf.py.

Reply:
That makes sense. I would say that most new attacks are implemented in `attacks.py` unless they rely on an auxiliary non-tensorflow component or they rely on an existing class that is used by the CleverHans attack class. This is at least how I drew the line so far when merging code. Let me know if you have any suggestions